### PR TITLE
Feature/ Toggle Event Types On/Off

### PR DESCRIPTION
**Pull request by mattchx**
**This feature is fully functional!** 
_Here are the details:_
1. styled Event Cards to create an on/off switch which makes card link disabled
2. added `active` (boolean) to EventType schema
3. created put request inside `<EventCard>` which sends boolean value and the event's unqiue id
4. create a new route in `eventAPI.js` which updates `active` to true or false in DB
5.  designed `<EventNotActivePage>`
6.  invalid page is displayed based on the active status of the event inside the `<Scheduler>` page

In addition: 
I may make refinements to the styling. But overall, I am very happy with the result. 
Please let me know if you have any suggestions!

### CODE REVIEW

File: client/src/components/UserEvents/EventCard.js
Comment (WhoaNellie) [inline client/src/components/UserEvents/EventCard.js:36]: 😃

File: client/src/components/SelectTabs/EventTypesTab.js
Comment (WhoaNellie) [inline client/src/components/SelectTabs/EventTypesTab.js]: If you could go through and make sure there are no console warnings from your code after you get the changes from my merge before you merge that would be great!

File: client/src/pages/Scheduler.js
Comment (WhoaNellie) [inline client/src/pages/Scheduler.js]: Could you use the eventDetails state and add an active key to it?

File: client/src/components/UserEvents/EventCard.js
Comment (alvyjudy) [inline client/src/components/UserEvents/EventCard.js]: I wonder if it's better to put this line (`setEventActive()`) into the `.then` block during our development. That way we guarantee that the request is completed before changing the UI. If this route goes wrong, it might be hard to notice on the front end.

It's not really a pressing issue though and we can leave it as it is.

**Review by alvyjudy — APPROVED** [review]
Looks really good nice work!

File: client/src/components/SelectTabs/EventTypesTab.js
Comment (dproc96) [inline client/src/components/SelectTabs/EventTypesTab.js]: Be sure to remove these console.logs

File: client/src/components/UserEvents/EventCard.js
Comment (dproc96) [reply client/src/components/UserEvents/EventCard.js]: I agree with this idea, it would be good to know if something went wrong because otherwise you could end up having the exact opposite state as the db

File: client/src/components/UserEvents/EventGrid.js
Comment (dproc96) [inline client/src/components/UserEvents/EventGrid.js]: This can be removed

File: client/src/pages/Scheduler.js
Comment (dproc96) [inline client/src/pages/Scheduler.js:82]: I'm seeing a flash of the scheduler page before being redirected, maybe you could use an isLoading state to wait until you know the event is active or not

File: server/db/models/EventType.js
Comment (dproc96) [inline server/db/models/EventType.js]: I think it would make more sense for user experience if the default value for this is true

**Review by dproc96 — COMMENTED** [review]
Overall I love the clean approach to this feature, just a couple of minor notes I added on to Nellie and Di's thoughts

File: client/src/components/UserEvents/EventCard.js
Comment (sina-jamshidi) [inline client/src/components/UserEvents/EventCard.js:99]: 👍

File: client/src/pages/EventNotActivePage.js
Comment (sina-jamshidi) [inline client/src/pages/EventNotActivePage.js:83]: Nice!

**Review by sina-jamshidi — APPROVED** [review]
This looks good to merge once comments are addressed!

File: client/src/pages/Scheduler.js
Comment (WhoaNellie) [reply client/src/pages/Scheduler.js:82]: We also have this problem with the onboarding page flashing before it detects that you've been onboarded, so whatever solution you end up doing for this we should steal for the onboarding page!

File: server/db/models/EventType.js
Comment (mattchx) [reply server/db/models/EventType.js]: Sure, I did that originally but removed it.

File: client/src/components/UserEvents/EventCard.js
Comment (mattchx) [reply client/src/components/UserEvents/EventCard.js]: I was having an async issue with setState so I did it like this but I knew it is not optimal! I can make the changes!

File: client/src/pages/Scheduler.js
Comment (mattchx) [reply client/src/pages/Scheduler.js]: Yes, that's valid. But I think for simplicity I will keep it as is unless there is a performance issue.

File: client/src/pages/Scheduler.js
Comment (mattchx) [reply client/src/pages/Scheduler.js:82]: @WhoaNellie I corrected this and I think solution is effective and simple. Check it out.

**Pull request merged**
